components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApiProductsApiServiceImpl.java

```diff
@@ -1039,4 +1047,98 @@ public Response updateAPIProductDeployment(String apiProductId, String deploymen
+                RestApiUtil.handleAuthorizationFailure("Authorization failure while retrieving the " +
+                        "lifecycle history of the API Product with id : " + apiProductId, e, log);
+            } else {
+                String errorMessage = "Error while retrieving the lifecycle history of  API Product with id : "
+                        + apiProductId;
+                RestApiUtil.handleInternalServerError(errorMessage, e, log);
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public Response getAPIProductLifecycleState(String apiProductId, String ifNoneMatch,
+                                                MessageContext messageContext) throws APIManagementException {
+
+        String organization = RestApiUtil.getValidatedOrganization(messageContext);
+        LifecycleStateDTO lifecycleStateDTO = getLifecycleState(apiProductId, organization);
+        return Response.ok().entity(lifecycleStateDTO).build();
+    }
+
+    private LifecycleStateDTO getLifecycleState(String apiProductId, String organization) {
+
+        try {
+            APIProductIdentifier productIdentifier = APIMappingUtil.getAPIProductIdentifierFromUUID(apiProductId,
+                    organization);
+            return PublisherCommonUtils.getLifecycleStateInformation(productIdentifier, organization);
+        } catch (APIManagementException e) {
+            if (RestApiUtil.isDueToResourceNotFound(e) || RestApiUtil.isDueToAuthorizationFailure(e)) {
+                RestApiUtil.handleResourceNotFoundError(RestApiConstants.RESOURCE_API_PRODUCT, apiProductId, e, log);
+            } else if (isAuthorizationFailure(e)) {
+                RestApiUtil.handleAuthorizationFailure("Authorization failure while retrieving the lifecycle " +
+                        "state information of API Product with id : " + apiProductId, e, log);
+            } else {
+                String errorMessage = "Error while retrieving the lifecycle state information of the API Product with" +
+                        " " + "id : " + apiProductId;
+                RestApiUtil.handleInternalServerError(errorMessage, e, log);
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public Response deleteAPIProductLifecycleStatePendingTasks(String apiProductId, MessageContext messageContext)
+            throws APIManagementException {
+
+        try {
+            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+            APIProductIdentifier productIdentifier = APIUtil.getAPIProductIdentifierFromUUID(apiProductId);
+            if (productIdentifier == null) {
+                throw new APIMgtResourceNotFoundException("Couldn't retrieve existing API Product with UUID: "
+                        + apiProductId, ExceptionCodes.from(ExceptionCodes.API_PRODUCT_NOT_FOUND, apiProductId));
+            }
+            apiProvider.deleteWorkflowTask(apiProductId, true);
```

Comment:
since we retrieve the APIProductIdentifier shall we have two methods for APIProductIdentifier and APIIdentifier.

Reply:
Fixed in c26fb7c